Add Module.md to Privacy stdlib (#14349)
ballerina-platform/ballerina-lang: stdlib/privacy/src/main/ballerina/privacy/Module.md

## Module overview

> Privacy module acts as the foundation of making Ballerina programming

Reviewer: ```suggestion
The privacy module acts as the foundation in making the Ballerina a privacy-aware programming
```

>  language privacy aware. The module provides required utility functions

Reviewer: ```suggestion
 language. The module provides the required utility functions
```

>  that allow pseudonymization, depseudonymization and deletion of

Reviewer: ```suggestion
 that allow pseudonymization, de-pseudonymization, and deletion of
```

>  personally identifiable information (PII).

Reviewer: ```suggestion
 Personally Identifiable Information (PII).
```

A pseudonymized identifier will be issued by the `pseudonymize` function
 for a provided PII. The actual PII value will be stored in the selected
 pluggable `PII Store`. The issued identifier can be used to represent
 the particular PII in other locations. The `depseudonymize` function

>  can be used to get the actual PII value, give the pseudonymized

Reviewer: ```suggestion
 can be used to obtain the actual PII value and give the pseudonymized
``

>  identifier. PII can be deleted from the storage using the `delete`

Reviewer: ```suggestion
 identifier. PII can be deleted from the store using the `delete`
```

Instead of "store", we can also say "the storage mechanism".

 function.

## Samples

```ballerina
import ballerina/h2;
import ballerina/io;
import ballerina/privacy;
import ballerina/sql;

h2:Client testDB = new({
    path: "./H2PIIStore/",
    name: "TestDBH2",
    username: "SA",
    password: "",
    poolOptions: { maximumPoolSize: 1 }
});

final string TABLE_NAME = "PII_STORE";
final string ID_CLOUMN = "id";
final string PII_COLUMN = "pii";

public function main(string pii) returns error? {

>     // Create database structure in H2 PII store.

Reviewer: ```suggestion
    // Creates the database structure in the H2 PII store.
```

    var creationStatus = testDB->update("CREATE TABLE IF NOT EXISTS PII_STORE (ID VARCHAR(300) NOT NULL, PII VARCHAR(300) NOT NULL, PRIMARY KEY (ID))");
    if (creationStatus is sql:UpdateResult) {
        io:println("PII to be persisted: " + pii);

>         // Create a storage that uses a H2 database to persist personally identifiable information (PII).

Reviewer: ```suggestion
        // Creates a storage that uses an H2 database to persist Personally Identifiable Information (PII).
```

Can we say "store" or "storage mechanism instead of "store"?

        privacy:H2PiiStore piiStore = new(testDB, TABLE_NAME, ID_CLOUMN, PII_COLUMN);

>         // Store PII information in the storage and take a pseudonymized identifier (UUID) representing stored data.

Reviewer: ```suggestion
        // Store PII information in the storage and take a pseudonymized identifier (UUID) representing the stored data.
```
Can we say "store" or "storage mechanism instead of "store"?

        var pseudonymizedPii = privacy:pseudonymize(piiStore, pii);
        if (pseudonymizedPii is string) {
            io:println("Pseudonymized identifier: " + pseudonymizedPii);

>             // Read the PII by providing the pseudonymized identifier.

Reviewer: ```suggestion
            // Reads the PII by providing the pseudonymized identifier.
```

            var depseudonymizedPii = privacy:depseudonymize(piiStore, pseudonymizedPii);
            if (depseudonymizedPii is string) {
                io:println("Deseudonymized value: " + depseudonymizedPii);
            } else {
                return depseudonymizedPii;
            }

>             // Delete the PII by providing the pseudonymized identifier.

Reviewer: ```suggestion
            // Deletes the PII by providing the pseudonymized identifier.
```
